### DISCUSSION TITLE
feature request: map users to org using the `X-Grafana-Org-Id` header

### DISCUSSION THREAD

**Discussion by bbigras** [grafana/grafana#43381] [General]
I use oauth2-proxy to auth user and I use 1 vhost for each org.

It would be nice, on the use first login, to map the user to the org set by the `X-Grafana-Org-Id` header.

**Comment by UXabre** [comment]
Hi, actually, this has been proposed, implemented and approved before unfortunately being shot down due to "conflicting commercial interests".
Anyway, here's my repository with your request implemented: https://github.com/UXabre/grafana
Mind you that this has also been enhanced with the ability to set SuperAdmin via auth proxy

To use it, I'd recommend building the dockerfile and utilize that (I haven't found the time to automatically build this).

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
